### ISSUE TITLE
Warnings causing building frameworks to fail

### ISSUE THREAD

**Issue by egu002** [google/j2objc#682]
make frameworks fails now with many warnings (and warnings being treated as errors)
The warnings are:
  -Wconversion
  -Wc++98-compat-pedantic
  -Wc99-extensions
  -Wc++98-compat
  -Wdocumentation-unknown-command
  -Wsuper-class-method-mismatch
  -Wold-style-cast
  -Wused-but-marked-unused
  -Wstatic-in-inline

**Comment by tomball** [comment]
What version of Xcode are you running?

**Comment by egu002** [comment]
Xcode 7.2, although I also tried with 7.1 (at least, the 7.1 command line tools)

**Comment by lukhnos** [comment]
I can also reproduce this with Xcode 7.2 (Clang 7.0.2 / clang-700.1.81). I could make `make frameworks` work if I suppressed a few more warnings mentioned above in [`static_framework.mk`](https://github.com/google/j2objc/blob/82001ee1f1748a914f17b341142bb19ebd5b2af9/make/static_framework.mk#L72) (here without the `\` line continuations). I've also added `-std=gnu++11` to the mix:

```
# Silences warnings when -std=c11
-Wno-conversion
-Wno-documentation-unknown-command
-Wno-static-in-inline
-Wno-super-class-method-mismatch
-Wno-used-but-marked-unused

# Silences warnings when -std=c11, -std=c++11 (and -std=gnu++11)
-Wno-c++98-compat
-Wno-c++98-compat-pedantic
-Wno-c99-extensions
-Wno-old-style-cast 
```

I wonder if applying all those would over-suppress some genuine issues. There are also a few C++-specific things, such as the preference of `static_cast` over old-style cast. Turning off `-Wc++98-compat-pedantic` is likely legit, as it tells you that `long long` is not part of C++98, even though C++11 has it and we are using `-std=c++11`.

The other thing about `make frameworks` is that if you run the command a few times despite the failures, eventually everything will be built. The make target is a combined header file that gets created as the first step of the recipe, and so subsequent runs will skip the checks since the target now already exists. :)

**Comment by lukhnos** [comment]
I used this to clean the combined headers, probably should have made it part of the Makefile:

```
#!/bin/bash
rm -f ./guava/build_result/Guava.h
rm -f ./inject/build_result/JavaxInject.h
rm -f ./jre_emul/build_result/JRE.h
rm -f ./jsr305/build_result/JSR305.h
rm -f ./junit/build_result/JUnit.h
rm -f ./protobuf/build_result/ProtobufRuntime.h
rm -f ./testing/build_result/Mockito.h
```

**Comment by tomball** [comment]
"make clean" removes all build_result directories (or it should).

**Comment by tomball** [comment]
We should make the warning checks optional. The warnings themselves don't
catch any errors since they're only checking header declarations. The only
reason I added them was to get ahead of over-zealous developers who turn
all warnings on with each new Xcode/clang release, and then complain that
the generated headers aren't warning-free.

**Comment by egu002** [comment]
I believe this is fixed in 1.0

**Comment by tomball** [comment]
Right you are. :-)
